extensions/ql-vscode/docs/test-plan.md

```diff
@@ -0,0 +1,257 @@
+8. Can copy repository list
+- Option is not shown before all analyses have finished
+- Option is shown once all analyses are finished
+- Text is copied to clipboard
+- Text is a valid repository list
+9. Can export results
+- Option is not shown before all analyses have finished
+- Option is shown once all analyses are finished
+10. Can select to create gist
+- A gist is created
+- The first thing in the gist is a summary
+- Contains a file for each repository with results
+- A popup links you to the gist
+11. Can select to save as markdown
+- A directory is created on disk
+- Contains a summary file
+- Contains a file for each repository with results
+- A popup allows you to open the directory
+12. Can cancel analysis
+- Option is shown when analysis is in progress
+- Option is not shown once all analyses are finished
+- Option is not shown before action run has been started
+- Causes the actions run to be cancelled
+
+### 4. Results view
+1. Can open a results view
+- When variant analysis state is “pending”
+- When variant analysis state is “in_progress”
+- When variant analysis state is “succeeded”
+- When variant analysis state is “failed”
+- When variant analysis state is “canceled”
+2. Results view opens automatically
+- When starting variant analysis run
+- When vscode opens (if view was open when vscode was closed)
+3. Can copy repository list
+See section from query history.
+4. Can export results
+- Only includes repos that you have selected (also see section from query history)
+5. Can cancel analysis (see section from query history)
+6. Can open query file (see section from query history)
+7. Can open query text
+8. Can view logs when variant analysis in final state (succeeded/failed/cancelled)
+9. Can sort repos
+  - By name
+  - By results
+  - By stars
+  - By last commit
+10. Can filter repos
+11. Shows correct statistics
+  - Total number of results
+  - Total number of repositories
+  - Duration
+13. Can see live results
+  - Results appear in extension as soon as each query is completed
+14. Can view interpreted results (i.e. for a “problem” query)
+  - Can view non-path results
+  - Can view code paths for “path-problem” queries
+15. Can view raw results (i.e. for a non “problem” query)
+  - Renders a table
+16. Can see skipped repositories 
+  - Can see repos with no db in a tab
+    - Shown warning that explains the tab
+  - Can see repos with no access in a tab
+    - Shown warning that explains the tab
+  - Only shows tab when there are skipped repos
+17. Result downloads
+  - All results are downloaded automatically
+  - Download status is indicated by a spinner (Not currently any indication of progress beyond “downloading” and “not downloading”)
+  - Only 3 items are downloaded at a time
+  - Results for completed queries are still downloaded when
+    - Some but not all queries failed
+    - The variant analysis was cancelled after some queries completed
+
+### 5. Upgrading from non-live to live results
```

Comment:
This section the next one and the one after that can be deleted - they were only relevant before releasing live results.